src/libs/auth.js

```diff
@@ -167,10 +168,14 @@ authStore.subscribe(withErrorReporting('Error checking registration', async (sta
   const getRegistrationStatus = async () => {
     try {
       const { enabled } = await Ajax().User.getStatus()
-      return enabled ? 'registered' : 'disabled'
+      if (enabled) {
+        return state.acceptedTos ? userStatus.registeredWithTos : userStatus.registeredWithoutTos
```

Comment:
From discussions with @marctalbott, I think that it is impossible for us to get "enabled" from this endpoint and yet have `acceptedTos` be false (it would be a bug). However, to be very explicit and because we do store `acceptedTos` separately, I am including `acceptedTos` in the status check.

I am tempted to make this an assertion instead of having a separate state for something we don't expect. Curious to see what reviewers think….

Reply:
I think that `acceptTos` will always be true when `enabled` is true when the grace period is over. It turns out that the fact that `registrationStatus` could be `registered` when `acceptedTos` was false was because of the code in Registered.js, not because of this method.

I think the assertion is the better way to go, perhaps after the grace period has ended everywhere though. I could add it to the general terra-ui "cleanup Tos code" ticket.

Reply:
I misread your initial comment as "it is now **possible** for us..." 🤦 . At any rate, this all sounds good to me